File: app/src/main/java/pl/kyon/forgottenfridge/EditActivity.java
package pl.kyon.forgottenfridge;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.Snackbar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.DatePicker;
import android.widget.EditText;

import org.joda.time.LocalDate;
import org.joda.time.format.DateTimeFormat;
import org.joda.time.format.DateTimeFormatter;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public class EditActivity extends AppCompatActivity {

    int id;
    Context ctx;
    DBHelper db;
    EditText etName;
    EditText etAmount;
    DatePicker dpExpire;
    final static String TAG = "EditActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_edit);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        ctx = this;
        db = new DBHelper(this);
        etName = (EditText) findViewById(R.id.etName);
        etAmount = (EditText) findViewById(R.id.etAmount);
        dpExpire = (DatePicker) findViewById(R.id.dpExpire);

        Button btnSave = (Button)findViewById(R.id.btnSave);
        Button btnSaveTop = (Button)findViewById(R.id.btnSaveTop);

        Intent intent = getIntent();
        id = intent.getIntExtra("id", 0);
        if (id > 0) {
            getSupportActionBar().setTitle(getString(R.string.title_activity_edit));
            Item i = db.getOne(id);
            etAmount.setText(i.amount);
            etName.setText(i.name);
            try {
                DateTimeFormatter dtf = DateTimeFormat.forPattern("yyyy-MM-dd");
                LocalDate date = dtf.parseLocalDate(i.expire);
                dpExpire.updateDate(date.getYear(), date.getMonthOfYear() - 1, date.getDayOfMonth());
            } catch (Exception e) {}

        } else {
            getSupportActionBar().setTitle(getString(R.string.title_activity_new));
        }
        View.OnClickListener clSave = new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                LocalDate date = new LocalDate()
                        .withYear(dpExpire.getYear())
                        .withMonthOfYear(dpExpire.getMonth() + 1)
                        .withDayOfMonth(dpExpire.getDayOfMonth());
                Log.v(TAG, date.toString());
                if (id > 0) {
                    db.edit(id, etName.getText().toString(), etAmount.getText().toString(), date.toString());
                } else {
                    db.add(etName.getText().toString(), etAmount.getText().toString(), date.toString());
                }
                onBackPressed();
            }
        };
        btnSave.setOnClickListener(clSave);
        btnSaveTop.setOnClickListener(clSave);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_edit, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();

        if (id == R.id.action_delete) {
            if (this.id > 0) {
                db.delete(this.id);
            }
            onBackPressed();
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

}

File: app/src/main/java/pl/mirkoczat/forgottenfridge/EditActivity.kt
package pl.mirkoczat.forgottenfridge

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.view.Menu
import android.view.MenuItem
import android.widget.Button
import android.widget.EditText
import com.codetroopers.betterpickers.calendardatepicker.CalendarDatePickerDialogFragment
import com.rollbar.android.Rollbar
import org.jetbrains.anko.AnkoLogger
import org.jetbrains.anko.find
import org.jetbrains.anko.info
import org.jetbrains.anko.onClick
import org.joda.time.DateTime
import java.util.*

const val DATE_FORMAT = "y-MM-dd"

class EditActivity: AppCompatActivity(), AnkoLogger {

    var id = 0
    var db: DBHelper? = null
    var etName: EditText? = null
    var etAmount: EditText? = null
    var etExpire: EditText? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_edit)
        setSupportActionBar(findViewById(R.id.toolbar) as Toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        if (ROLLBAR_ENABLED)
            Rollbar.init(this, ROLLBAR_ID, ROLLBAR_ENV);

        etName = findViewById(R.id.etName) as EditText
        etAmount = findViewById(R.id.etAmount) as EditText
        etExpire = findViewById(R.id.etExpire) as EditText

        val btnExpire = findViewById(R.id.btnExpire) as Button
        btnExpire.onClick {
            val date = DateTime.parse(etExpire?.text.toString())
            CalendarDatePickerDialogFragment()
                    .setOnDateSetListener { calendarDatePickerDialogFragment, y, m, d ->
                        val dt = DateTime(y, m + 1, d, 0, 0).toString(DATE_FORMAT)
                        etExpire?.setText(dt)
                    }
                    .setFirstDayOfWeek(Calendar.MONDAY)
                    .setPreselectedDate(date.year, date.monthOfYear - 1, date.dayOfMonth)
                    .show(supportFragmentManager, "")
        }
        val btnSave = findViewById(R.id.btnSave) as Button
        val btnSaveTop = findViewById(R.id.btnSaveTop) as Button
        btnSave.onClick { saveAction() }
        btnSaveTop.onClick { saveAction() }

        db = DBHelper(this)
        id = intent.getIntExtra("id", 0)

        when (id) {
            0 -> {
                supportActionBar?.title = getString(R.string.title_activity_new)
                etExpire?.setText(DateTime.now().toString(DATE_FORMAT))
            }
            else -> {
                supportActionBar?.title = getString(R.string.title_activity_edit)
                val product = db?.getOne(id)

                etAmount?.setText(product?.amount)
                etName?.setText(product?.name)
                etExpire?.setText(product?.expire)
            }
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_edit, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem) = when (item.itemId) {
        R.id.action_delete -> {
            if (id > 0)
                db?.delete(id)
            onBackPressed()
            true
        }
        else -> super.onOptionsItemSelected(item)
    }

    fun saveAction() {
        val item = Product(id, etName?.text.toString(), etAmount?.text.toString(), etExpire?.text.toString())
        when {
            id > 0 -> db?.edit(id, item)
            else  -> db?.add(item, true)
        }
        onBackPressed()
    }
}
